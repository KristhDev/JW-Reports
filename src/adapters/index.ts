export * from './auth';
export * from './course';
export * from './preaching';
export * from './revisits';