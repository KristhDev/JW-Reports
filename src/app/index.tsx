import React from 'react';
import { Redirect } from 'expo-router';

import { useAuth } from '@auth';

export default function MainScreen() {
    const { state: { isAuthenticated } } = useAuth();

    if (isAuthenticated) return <Redirect href="/(app)/(tabs)/preaching" />;
    else return <Redirect href="/auth/login" />;
}