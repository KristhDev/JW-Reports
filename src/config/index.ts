export * from './env';
export * from './calendar';
export * from './supabase';