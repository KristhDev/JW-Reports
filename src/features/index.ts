export * from './auth';
export * from './courses';
export * from './hooks';
export * from './permissions';
export * from './preaching';
export * from './revisits';
export * from './status';
export * from './store';