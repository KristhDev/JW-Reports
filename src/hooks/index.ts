export { default as useAuth } from './useAuth';
export { default as useTheme } from './useTheme';