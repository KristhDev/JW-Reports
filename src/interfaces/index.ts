export * from './auth';
export * from './courses';
export * from './features';
export * from './permissions';
export * from './preaching';
export * from './revisits';
export * from './status';
export * from './theme';
export * from './ui';
