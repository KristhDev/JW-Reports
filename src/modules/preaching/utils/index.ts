export * from './constants';
export * from './report';