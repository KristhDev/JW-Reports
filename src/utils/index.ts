export * from './calcs';
export * from './constants';