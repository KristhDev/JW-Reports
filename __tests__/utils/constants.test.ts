
/* Utils */
import { PRECURSORS_OPTIONS, TABLE_PREACHING_HEADERS, THEME_OPTIONS, HOURS_REQUIREMENTS } from '../../src/utils';

describe('Test in Constants util', () => {
    it('should to match snapshot TABLE_PREACHING_HEADERS', () => {
        expect(TABLE_PREACHING_HEADERS).toMatchSnapshot();
    });

    it('should to match snapshot PRECURSORS_OPTIONS', () => {
        expect(PRECURSORS_OPTIONS).toMatchSnapshot();
    });

    it('should to match snapshot THEME_OPTIONS', () => {
        expect(THEME_OPTIONS).toMatchSnapshot();
    });

    it('should to match snapshot HOURS_REQUIREMENTS', () => {
        expect(HOURS_REQUIREMENTS).toMatchSnapshot();
    });
});