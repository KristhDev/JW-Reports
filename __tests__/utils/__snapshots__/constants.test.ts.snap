// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Constants util should to match snapshot PRECURSORS_OPTIONS 1`] = `
Array [
  Object {
    "label": "Ninguno",
    "value": "ninguno",
  },
  Object {
    "label": "Auxiliar",
    "value": "auxiliar",
  },
  Object {
    "label": "Regular",
    "value": "regular",
  },
  Object {
    "label": "Especial",
    "value": "especial",
  },
]
`;

exports[`Test in Constants util should to match snapshot TABLE_PREACHING_HEADERS 1`] = `
Array [
  "DÃ­a",
  "H/I",
  "H/F",
  "Pub",
  "Vid",
  "Rev",
]
`;

exports[`Test in Constants util should to match snapshot THEME_OPTIONS 1`] = `
Array [
  Object {
    "label": "Modo predeterminado",
    "value": "default",
  },
  Object {
    "label": "Modo claro",
    "value": "light",
  },
  Object {
    "label": "Modo oscuro",
    "value": "dark",
  },
]
`;
